### ISSUE TITLE
[HOLD for payment 2024-05-20] [$250] Whispered messages showing to all users temporarily when scrolling

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#41464]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @zsgreenwald 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1714505816123539

## Action Performed:
1. Open any public room
2. Scroll the page up and down

## Expected Result:
only the whispered messages intended for you are displayed
## Actual Result:
whispered messages intended to someone else are displayed
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/46b3c3f4-6813-4713-9e0b-72297dd8b47f

https://github.com/Expensify/App/assets/38435837/9de8ca5f-86c9-4e4b-9853-334528ac0663

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01afb8dc7580ba1380</li>
        <li>Upwork Job ID: 1787563116814917632</li>
        <li>Last Price Increase: 2024-05-06</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sonialiap</details>

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by marcaaron** [comment]
Chatting about this in Slack and it seems like this can be fixed Externally. We need to deprecate `reportAction.whisperedToAccountIDs` and use `reportAction.message.whisperedTo` instead (which as far as we can tell is always the same).

We are deprecating `reportAction.originalMessage` in [this issue](https://github.com/Expensify/App/issues/39797). One way to migrate this would be something like:

```js
if (isObject(reportAction.message)) {
    whisperedToAccountIDs = reportAction.message?.whisperedTo ?? [];
} else {
    whisperedToAccountIDs = reportAction?.originalMessage?.whisperedTo ?? [];
}
```

We can come back later to remove the `originalMessage` fallback once the Internal part of the linked issue is done.

There is some Internal backend cleanup to do as well since we have `whisperedToAccountIDs` being sent for at least one spot in Auth [here](https://github.com/Expensify/Auth/blob/ce2bede820633de816001c10efa26a15a62f05b3/auth/command/EditMoneyRequest.cpp#L382-L384).

@luacmartins also noticed there are som missing keys when sending updates via Auth (which should be fixed, but unclear if causing any specific problem for New Dot). But maybe we can create a separate issue for that since it does not seem related to this bug.

**Comment by marcaaron** [comment]
Working on this as we need it quick - still a front end change for now.

**Comment by sonialiap** [comment]
@jayeshmangwani Upwork is not letting me send an offer so I've invited you to the job. Could you submit a proposal - it does not need to be an actual proposal, just something so that I can accept and send you an offer?

**Comment by jayeshmangwani** [comment]
@sonialiap Yes sure, I've Applied

**Comment by sonialiap** [comment]
Payment summary:
@jayeshmangwani $250 - paid ✔️ 

Please complete the [checklist](https://github.com/Expensify/App/issues/41464#issuecomment-2106925472) :)

**Comment by jayeshmangwani** [comment]
- [ ] [@jayeshmangwani] The PR that introduced the bug has been identified. Link to the PR: N/A
- [ ]  [@jayeshmangwani] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- [ ]  [@jayeshmangwani] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **IMO, this bug is an improvement; rather than showing whisper to all users, we decided to show the respective user.**
- [x]  [@jayeshmangwani] Determine if we should create a regression test for this bug.
- [x]  [@jayeshmangwani] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

 
 ## Regression Test Proposal

1. As User A create a workspace
2. Create a public workspace room
3. Give the workspace room a description
4. As User B access the workspace room
5. As User B - Verify that the whisper message displays
6. As User A verify that no whisper messages are displayed.

Do we agree 👍 or 👎

@sonialiap
